.github/workflows/do-release.yml

```diff
@@ -0,0 +1,158 @@
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          draft: true
+          prerelease: false
+      - uses: actions/download-artifact@v1
+        with:
+          name: sgrep-osx-${{ github.sha }}
+          path: sgrep-osx
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+      - name: help
+        run: |
+          pwd;
+          ls;
+          find .;
+      - name: Upload Release Asset
+        id: upload-release-asset-osx
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./sgrep-osx/artifacts.zip
+          asset_name: sgrep-osx-${{ steps.get_version.outputs.VERSION }}.zip
+          asset_content_type: application/zip
+      - uses: actions/download-artifact@v1
+        with:
+          name: sgrep-ubuntu-16.04-${{ github.sha }}
+          path: sgrep-ubuntu
+      - name: Upload Release Asset
+        id: upload-release-asset-ubuntu
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./sgrep-ubuntu/artifacts.tar.gz
+          asset_name: sgrep-ubuntu-1604-${{ steps.get_version.outputs.VERSION }}.tar.gz
+          asset_content_type: application/gzip
+  release-osx:
+    name: Build the OSX binaries
+    runs-on: macos-latest
+    # TODO: this is very similar to the mac-build workflow. Convert to an action
+    steps:
+      - name: Install System Deps
+        run: brew install opam pkg-config coreutils
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: OPAM setup
+        run: |
+          opam init --no-setup --bare;
+          opam switch create 4.07.1;
+          opam switch 4.07.1;
+      - name: OPAM install Deps
+        run: eval $(opam env); opam install -y reason dune ocamlfind camlp4 num ocamlgraph json-wheel conf-perl yaml
+      - name: Install submodules
+        run: git submodule update --init --recursive
+      - name: Install pfff
+        run: eval $(opam env) && opam install -y ./pfff
+      - name: Install and test sgrep
+        run: eval $(opam env) && cd sgrep && make all && make test && make install
+      - name: Prepare artifacts
+        run: |
+          mkdir -p artifacts;
+          cp ./sgrep/_build/default/bin/main_sgrep.exe artifacts/sgrep;
+          zip -r artifacts.zip artifacts
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: sgrep-osx-${{ github.sha }}
+          path: artifacts.zip
+  release-ubuntu-16-04:
+    # TODO: this is very similar to the ubuntu-build workflow.
+    # convert to github action to DRY
```

Comment:
How much work is it to DRY in this PR instead of a follow up PR?

Reply:
probably 1-2h? Not sure I'm also not 100% we want them to be exactly the same? eg. maybe there's a faster way to run tests

Reply:
actually turns out this is literally impossible for OSX tasks 🙄

Reply:
I'm worried if they arent the same then this test might pass but the actual binary built by the other job is broken